crates/transaction-pool/src/pool/blob.rs

```diff
@@ -237,6 +314,6 @@ impl PartialOrd<Self> for BlobOrd {
 
 impl Ord for BlobOrd {
     fn cmp(&self, other: &Self) -> Ordering {
-        other.submission_id.cmp(&self.submission_id)
+        other.priority.total_cmp(&self.priority)
```

Comment:
oh I now remember why we had the submission id, this is used as a guarantee so that all values are unique, because now we lose entries if a tx has the same priority...

Reply:
ohh right, yeah we would lose elements now with the btreeset. BinaryHeap does seem like it would work